crates/orderbook/src/order_validation.rs

```diff
@@ -572,6 +585,40 @@ fn is_order_outside_market_price(order: &OrderData, quote: &Quote) -> bool {
     order.sell_amount.full_mul(quote.buy_amount) < quote.sell_amount.full_mul(order.buy_amount)
 }
 
+fn convert_signing_scheme_into_quote_signing_scheme(
+    scheme: SigningScheme,
+    order_placement_via_api: bool,
+) -> Result<QuoteSigningScheme, ValidationError> {
+    match order_placement_via_api {
+        true => match scheme {
```

Comment:
This match statement and the one in the function below would be more readable by combing the inner match with the outer:
```rust
match (order_placement_via_api, scheme) {
    (true, SigningScheme::Presign) => todo!(),
    ...
}